src/Microsoft.AspNetCore.Routing/Matchers/MatcherEndpoint.cs

```diff
@@ -30,14 +32,15 @@ public sealed class MatcherEndpoint : Endpoint
 
             Invoker = invoker;
             Template = template;
+            RouteTemplate = TemplateParser.Parse(template);
             Values = new RouteValueDictionary(values);
             Order = order;
         }
 
         public int Order { get; }
         public Func<RequestDelegate, RequestDelegate> Invoker { get; }
         public string Template { get; }
-
+        public RouteTemplate RouteTemplate { get; }
```

Comment:
I have a few reservations about exposing this type prominently because it's kinda gross... I don't think it's blocking for now.

Reply:
At least add a comment to say this needs to be review